spark-connector/build.gradle.kts

```diff
@@ -14,26 +15,35 @@ repositories {
 
 val scalaVersion: String = project.properties["scalaVersion"] as? String ?: extra["defaultScalaVersion"].toString()
 val sparkVersion: String = libs.versions.spark.get()
+val sparkMajorVersion: String = extra["sparkMajorVersion"].toString()
 val icebergVersion: String = libs.versions.iceberg.get()
 val kyuubiVersion: String = libs.versions.kyuubi.get()
 val scalaJava8CompatVersion: String = libs.versions.scala.java.compat.get()
 
 dependencies {
-  implementation(project(":api"))
   implementation(project(":clients:client-java-runtime", configuration = "shadow"))
-  implementation(project(":common"))
-  implementation(libs.bundles.log4j)
   implementation(libs.guava)
   implementation("org.apache.iceberg:iceberg-spark-runtime-3.4_$scalaVersion:$icebergVersion")
```

Comment:
You would better use Spark Major version here, too.

Reply:
done